src/Examples/Flowsheets/ngcc/ngcc_soec_costing.py

```diff
@@ -0,0 +1,792 @@
+
+    # accounts with flue gas flowrate to HRSG as the process parameter
+    hrsg_flow_accounts = ["7.6"]
+
+    m.fs.b12 = pyo.Block()
+
+    get_PP_costing(
+        m.fs.b12, hrsg_flow_accounts, m.fs.hrsg_gas_flow[0], "acfm", 6)
+
+    # accounts with steam turbine gross power as the process parameter
+    st_power_accounts = ["8.1", "8.2", "8.5", "14.3"]
+
+    m.fs.b13 = pyo.Block()
+
+    get_PP_costing(
+        m.fs.b13, st_power_accounts, m.fs.steam_turbine_gross_power[0], "kW", 6)
+
+    # accounts with condenser duty as the process parameter
+    condenser_duty_accounts = ["8.3"]
+
+    m.fs.b14 = pyo.Block()
+
+    get_PP_costing(
+        m.fs.b14, condenser_duty_accounts, m.fs.condenser_duty[0], "MMBtu/hr", 6)
+
+    # accounts with cooling tower duty as the process parameter
+    cooling_tower_accounts = ["9.1"]
+
+    m.fs.b15 = pyo.Block()
+
+    get_PP_costing(
+        m.fs.b15, cooling_tower_accounts, m.fs.cooling_water_duty[0], "MMBtu/hr", 6)
+
+    # accounts with circulating water flowrate as the process parameter
+    cooling_water_accounts = ["9.2", "9.3", "9.4", "9.6", "9.7", "14.5"]
+
+    m.fs.b16 = pyo.Block()
+
+    get_PP_costing(
+        m.fs.b16, cooling_water_accounts, m.fs.cooling_water_flow[0], "gpm", 6)
+
+    # accounts with total plant gross power as the process parameter
+    gross_power_accounts = ["11.1", "11.7", "11.9",
+                            "13.1", "13.2", "13.3",
+                            "14.4", "14.7", "14.8", "14.9", "14.10"]
+
+    m.fs.b17 = pyo.Block()
+
+    get_PP_costing(
+        m.fs.b17, gross_power_accounts, m.fs.gross_power[0], "kW", 6)
+
+    # transformer costs
+    gross_power_accounts_2 = ["11.8"]
+
+    m.fs.b18 = pyo.Block()
+
+    get_PP_costing(
+        m.fs.b18, gross_power_accounts_2, m.fs.gross_power[0], "kW", 6)
+
+    # accounts with auxilliary load as the process parameter
+    aux_load_accounts = ["11.2", "11.3", "11.4", "11.5", "11.6",
+                         "12.1", "12.2", "12.3", "12.4", "12.5", "12.6",
+                         "12.7", "12.8", "12.9"]
+
+    m.fs.b19 = pyo.Block()
+
+    get_PP_costing(m.fs.b19, aux_load_accounts, m.fs.aux_load[0], "kW", 6)
+
+    ###########################################################################
+    # SOEC CAPITAL COSTING
+
+    # soec modules
+    m.fs.soec.soec_module.costing = pyo.Block()
+
+    @m.fs.soec.soec_module.costing.Expression()
+    def total_plant_cost(c):
+        extra_installed_area = 1.10  # accounts for cell degradation
+        return 60.87 * m.fs.soec.soec_module.number_cells * extra_installed_area / 1e6
+
+    # high temperature soec trim heaters
+    for heater in [m.fs.soec.feed_heater, m.fs.soec.sweep_heater]:
+        heater.costing = pyo.Block()
+
+        @heater.costing.Expression()
+        def total_plant_cost(b):
+            U = 100 * pyo.units.W / pyo.units.m ** 2 / pyo.units.K
+            DT = 50 * pyo.units.K
```

Comment:
Might be better to revise this down to 20 K. The U value here is also optimistic (Jinliang's model predicted ~40-60 at 1.2 bar, without considering an optimal design), but is in line with what we're using for the rest of the report.

Reply:
I made a mistake on my objective function scaling in my last commit, so I'll fix that and update this, and hopfully the next commit will be good.